FrmMainApp.cs

```diff
@@ -613,128 +494,236 @@ public class MapGpsCoordinates
     }
 
+
     /// <summary>
-    ///     Handles the navigation to a coordinate on the map. Replaces hard-coded values w/ user-provided ones
-    ///     ... and executes the navigation action.
+    ///     Parses the tbx_lat and tbx_lng text boxes.
+    ///     If the contents is a valid double, returns touple (lat, lng)
+    ///     with values as string and dec separator ".".
+    ///     Otherwise default "0" is returned for both.
     /// </summary>
-    private void NavigateMapGo()
+    private (string, string) parseLatLngTextBox()
     {
-        Logger.Debug(message: "Starting");
+        Logger.Trace(message: "Starting parseLatLngTextBox ...");
 
-        string htmlCode = "";
-        HelperStatic.HtmlAddMarker = "";
-        double? dblMinLat = default;
-        double? dblMinLng = default;
-        double? dblMaxLat = default;
-        double? dblMaxLng = default;
+        // Default values if text field is empty
+        string LatCoordinate = "0";
+        string LngCoordinate = "0";
 
-        // lazy
-        string strLatCoordinate = "0";
-        string strLngCoordinate = "0";
+        // Get txtbox contents
+        string strLatCoordinate = "";
+        string strLngCoordinate = "";
+        if (tbx_lat.Text != null) strLatCoordinate = tbx_lat.Text.Replace(oldChar: ',', newChar: '.');
+        if (tbx_lng.Text != null) strLngCoordinate = tbx_lng.Text.Replace(oldChar: ',', newChar: '.');
 
+        // Check if it's a valid double -> otherwise defaults above
         try
         {
-            htmlCode = File.ReadAllText(path: Path.Combine(path1: ResourcesFolderPath, path2: "map.html"));
+            Logger.Trace(message: "parseLatLngTextBox");
+            double parsedLat;
+            double parsedLng;
+            if (double.TryParse(s: strLatCoordinate, style: NumberStyles.Any,
+                provider: CultureInfo.InvariantCulture, result: out parsedLat) &&
+                double.TryParse(s: strLngCoordinate, style: NumberStyles.Any,
+                provider: CultureInfo.InvariantCulture, result: out parsedLng))
+            {
+                LatCoordinate = strLatCoordinate;
+                LngCoordinate = strLngCoordinate;
+                Logger.Trace(message: "parseLatLngTextBox OK - LatCoordinate: " + strLatCoordinate + " - LngCoordinate: " + strLngCoordinate);
+            }
         }
         catch (Exception ex)
         {
-            Logger.Fatal(message: "ReadAllText map.html Error:" + ex.Message);
-            MessageBox.Show(text: HelperStatic.GenericGetMessageBoxText(messageBoxName: "mbx_FrmMainApp_ErrorNavigateMapGoHTMLCode") + ex.Message, caption: "Error", buttons: MessageBoxButtons.OK, icon: MessageBoxIcon.Error);
+            Logger.Fatal(message: "Error: " + ex.Message);
+            MessageBox.Show(text: HelperStatic.GenericGetMessageBoxText(messageBoxName: "parseLatLngTextBox_ParseDouble") + ex.Message, caption: "Error", buttons: MessageBoxButtons.OK, icon: MessageBoxIcon.Error);
```

Comment:
`messageBoxName: "mbx_FrmMainApp_ErrorNavigateMapGoHTMLCode") `needs to stay in otherwise the code won't find it. basically if you look in the language files all messagebox messages have an mbx_somethingsomething logic so it's not the "literal" "parseLatLngTextBox_ParseDouble" that we are after but a line in the language file that starts w/ mbx_ and then has a message attached to it.